axis_utils/include/axis_utils2.inc

```diff
@@ -351,119 +182,88 @@ end subroutine axis_edges
     !        lon =      0 1 2 3 4 5  ...  358 359; lon_strt = 3 ==>
     !        tranlon =  3 4 5 6 7 8  ...  359 360 361 362; istrt = 4
 
-    real, intent(inout), dimension(:) :: lon
-    real, intent(in) :: lon_start
-    integer, intent(out) :: istrt
+    real(kind=FMS_AU_KIND_), intent(inout), dimension(:) :: lon
+    real(kind=FMS_AU_KIND_), intent(in)                  :: lon_start
+    integer, intent(out)                                 :: istrt
 
 
-    integer :: len, i
-    real :: lon_strt, tmp(size(lon(:))-1)
+    integer                 :: len, i
+    real(kind=FMS_AU_KIND_) :: lon_strt, tmp(size(lon(:))-1)
 
     len = size(lon(:))
 
-    do i=1,len
+    do i = 1, len
        lon(i) = lon_in_range(lon(i),lon_start)
     enddo
 
-    istrt=0
-    do i=1,len-1
+    istrt = 0
+    do i = 1,len-1
        if (lon(i+1) < lon(i)) then
-          istrt=i+1
+          istrt = i+1
           exit
        endif
     enddo
 
     if (istrt>1) then ! grid is not monotonic
-       if (abs(lon(len)-lon(1)) < epsln) then
+       if (abs(lon(len)-lon(1)) < real(epsln, r8_kind)) then
           tmp = cshift(lon(1:len-1),istrt-1)
           lon(1:len-1) = tmp
-          lon(len) = lon(1)
+          lon(len)     = lon(1)
        else
           lon = cshift(lon,istrt-1)
        endif
+
        lon_strt = lon(1)
        do i=2,len+1
-          lon(i) = lon_in_range(lon(i),lon_strt)
+          lon(i)   = lon_in_range(lon(i),lon_strt)
           lon_strt = lon(i)
        enddo
     endif
 
     return
-  end subroutine tranlon
+  end subroutine TRANLON_
 
-  !>     nearest_index = index of nearest data point within "array" corresponding to
-  !!            "value".
-  !!
-  !!     inputs:
-  !!
-  !!     value  = arbitrary data...same units as elements in "array"
-  !!     array  = array of data points  (must be monotonically increasing)
-  !!
-  !!     output:
-  !!
-  !!     nearest_index =  index of nearest data point to "value"
-  !!             if "value" is outside the domain of "array" then nearest_index = 1
-  !!             or "ia" depending on whether array(1) or array(ia) is
-  !!             closest to "value"
-  !!
-  !!             note: if "array" is dimensioned array(0:ia) in the calling
-  !!                   program, then the returned index should be reduced
-  !!                   by one to account for the zero base.
-  !!
-  !!     example:
-  !!
-  !!     let model depths be defined by the following:
-  !!     parameter (km=5)
-  !!     dimension z(km)
-  !!     data z /5.0, 10.0, 50.0, 100.0, 250.0/
-  !!
-  !!     k1 = nearest_index (12.5, z, km)
-  !!     k2 = nearest_index (0.0, z, km)
-  !!
-  !!     k1 would be set to 2, and k2 would be set to 1 so that
-  !!     z(k1) would be the nearest data point to 12.5 and z(k2) would
-  !!   be the nearest data point to 0.0
-  !!
-  !!   @return real frac_index
```

Comment:
Why delete this documentation? was it moved anywhere else

Reply:
This comment block is a description of the routine nearest_index. It looks like it somehow got copied & pasted above frac_index as well as nearest_index so I just removed it's incorrect placed repeat